src/main/java/kitchenpos/products/application/ProductService.java

```diff
@@ -32,38 +35,28 @@ public ProductService(
     }
 
     @Transactional
-    public Product create(final Product request) {
-        final BigDecimal price = request.getPrice();
-        if (Objects.isNull(price) || price.compareTo(BigDecimal.ZERO) < 0) {
-            throw new IllegalArgumentException();
-        }
-        final String name = request.getName();
-        if (Objects.isNull(name) || purgomalumClient.containsProfanity(name)) {
-            throw new IllegalArgumentException();
-        }
-        final Product product = new Product();
-        product.setId(UUID.randomUUID());
-        product.setName(name);
-        product.setPrice(price);
+    public Product create(final ProductModel request) {
```

Comment:
create
```suggestion
    public Product create(final Bigdeciaml price, final String name) {
```
dto를 presentation layer 까지만 사용하는 것에 대해서 어떻게 생각히시나요?

Reply:
1.dto 자체를 presentation layer 까지만 사용 하는것에 관하여 말씀 하시는 건지, 
2. 소스 기준 ui, application package에서 서로 다른 dto를 사용하는 것에 대하여 물어 보시는건지 조금 헷갈리기는 하는데

1.번을 말씀하시는 거라면 dto, domain model을 사용하지 않고 어떻게 데이터를 주고 받는지 궁금합니다.
2.번을 말씀하시는 거라면 회사에서 개발한다는 가정으로 팀 내에서 의견 조율할 문제라 생각합니다. dto의 갯수가 많아져도 내부 -> 외부 
의존성을 완전히 제거 할 것인지, 의존성이 생기더라도 개발 편의를 위해 dto를 하나로 사용할 것인지에 대한 의견 조율 문제라 생각합니다.

리뷰어님은 어떻게 생각하시나요? 그리고 질문의 정확한 의도가 궁금합니다.

Reply:
1번에 가까운 질문이였던것 같습니다.
클라이언트에서 넘겨받은 DTO 를 서비스레이어에서도 사용한다면 서비스레이어가 클라이언트와 의존성을 가지게 되지 않을까 고민하고있습니다. 그래서 예시로 남겨드린것과 같이 필드를 분해해서 받으면 메소드의 재사용성도 좋아지지 않을까 생각합니다.